File: backend/src/routes/auth.routes.js
import { Router } from "express";
import {
    loginUser,
    logout,
    registerUser,
    validation,
} from "../controllers/auth.controller.js";
import { authValidation } from "../middlewares/auth.middleware.js";

export const authRouter = Router();

authRouter.route("/register").post(registerUser);
authRouter.route("/login").post(loginUser);
authRouter.route("/logout").post(logout);
authRouter.route("/validation").get(authValidation, validation)

File: backendFlask/src/routes/auth_routes.py
from flask import Blueprint
from controllers.auth_controller import register_user, login_user, logout, validation
from middlewares.auth_middleware import auth_validation

auth_bp = Blueprint("auth", __name__)

auth_bp.route("/register", methods=["POST"])(register_user)
auth_bp.route("/login", methods=["POST"])(login_user)
auth_bp.route("/logout", methods=["POST"])(logout)
auth_bp.route("/validation", methods=["GET"])(auth_validation(validation))
